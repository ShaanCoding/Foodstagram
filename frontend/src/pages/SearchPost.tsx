// import styles from "../styles/Search.module.css";
import { useParams } from 'react-router-dom'
import useAuth from '../api/util/useAuth'
import UseFeedQuery from '../api/UseFeedQuery'
import { Post } from '../components/post/Post'
import { useEffect, useState } from 'react'
import Spinner from '../components/common/Spinner'
import Fuse from 'fuse.js'


const SearchPost = () => {
  const [account, isLoading] = useAuth()
  const { searchStr } = useParams();
	const feedQuery = UseFeedQuery()
  const feeds = feedQuery.data?.data.posts
  let result: any = []
  let finalResult: any = []


  if (feeds !== undefined && feeds.length > 0 && searchStr !== undefined) {
    const fuse = new Fuse(feeds, {
      shouldSort: true,
      threshold: 0.5,
      keys: ["location_name"]
    })
    result = fuse.search(searchStr)
  }

  if (result.length) {
    result.forEach((item: any) => {
      finalResult.push(item.item)
    })
  }

  return (
    <>
      <h1 className='text-justify text-2xl px-48'>Search results for '{searchStr}'</h1>
      
      <div> 
        {//no results
          result === undefined || result.length == 0 && finalResult.length == 0 && (
            <h1 className='mx-auto block w-2/3 text-center mt-[200px] mb-[200px] text-xl text'>
              No results
            </h1>
          )
        }
        
        {//posts results
        finalResult !== undefined && finalResult.length > 0 &&
          finalResult.map((Object:any) => {
            return (
              <Post key={Object.post_id} post={Object} />
            )
          })
        }

      </div>     
    </>
  )
}

export default SearchPost