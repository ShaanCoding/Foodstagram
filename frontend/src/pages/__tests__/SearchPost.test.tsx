import { QueryClient, QueryClientProvider } from 'react-query'
import { BrowserRouter, Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import SearchPost from '../SearchPost'

const queryClient = new QueryClient()

it('The Search Post/Location page renders correctly', () => {
	const tree = renderer
		.create(
			<QueryClientProvider client={queryClient}>
				<BrowserRouter>
					<SearchPost />
				</BrowserRouter>
			</QueryClientProvider>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
