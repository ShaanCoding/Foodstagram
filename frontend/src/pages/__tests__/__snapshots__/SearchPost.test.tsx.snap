// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Search Post/Location page renders correctly 1`] = `
[
  <h1
    className="text-justify text-2xl px-48"
  >
    Search results for '
    '
  </h1>,
  <div
    className="w-full lg:w-1/2 flex justify-end items-center relative"
  />,
]
`;
